src/api-extension/api-extension.service.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable } from '@nestjs/common';
+import { CommerceToolsConnectorService } from '../commerceTools/commerce-tools-connector.service';
+import { ByProjectKeyRequestBuilder } from '@commercetools/platform-sdk/dist/declarations/src/generated/client/by-project-key-request-builder';
+import { Extension } from '@commercetools/platform-sdk';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class ApiExtensionService {
+  private client: ByProjectKeyRequestBuilder;
+  constructor(
+    private readonly ctConnector: CommerceToolsConnectorService,
+    private readonly configService: ConfigService,
+  ) {
+    this.client = this.ctConnector.getClient();
+  }
+
+  async list() {
+    const {
+      body: { results: extensions },
+    } = await this.client.extensions().get().execute();
+
+    return extensions;
+  }
+
+  async removeById(id: string) {
+    const extensions = await this.list();
+    const extensionToRemove = extensions.find(
+      (extension) => extension.id === id,
+    );
+
+    if (!extensionToRemove) {
+      return false;
+    }
+    const removedExtenison = (
+      await this.client
+        .extensions()
+        .withId({ ID: extensionToRemove.id })
+        .delete({ queryArgs: { version: extensionToRemove.version } })
+        .execute()
+    ).body;
+
+    return removedExtenison;
+  }
+
+  async removeByAttr(attr: 'key' | 'destination', value: string) {
+    const extensions = await this.list();
+    const extensionsToRemove = extensions.filter(
+      (extension) => extension[attr] === value,
+    );
+    if (!extensionsToRemove?.length) {
+      return false;
+    }
+
+    const removedExtensions: Extension[] = [];
+
+    for (const extension of extensionsToRemove) {
+      removedExtensions.push(
+        (
+          await this.client
+            .extensions()
+            .withId({ ID: extension.id })
+            .delete({ queryArgs: { version: extension.version } })
+            .execute()
+        ).body,
+      );
+    }
+
+    return removedExtensions;
+  }
+
```

Comment:
Consider adding return types in `removeByAttr`, `removeById`, `list`.

Reply:
maybe later